confmap/provider/googlesecretmanagerprovider/provider.go

```diff
@@ -0,0 +1,86 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+//go:generate mdatagen metadata.yaml
+
+package googlesecretmanagerprovider // import "github.com/open-telemetry/opentelemetry-collector-contrib/confmap/provider/googlesecretmanagerprovider"
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"sync"
+
+	secretmanager "cloud.google.com/go/secretmanager/apiv1"
+	gax "github.com/googleapis/gax-go/v2"
+	"github.com/googleapis/gax-go/v2/apierror"
+	"go.opentelemetry.io/collector/confmap"
+
+	"cloud.google.com/go/secretmanager/apiv1/secretmanagerpb"
+)
+
+type secretsManagerClient interface {
+	AccessSecretVersion(ctx context.Context, req *secretmanagerpb.AccessSecretVersionRequest, opts ...gax.CallOption) (*secretmanagerpb.AccessSecretVersionResponse, error)
+	Close() error
+}
+
+const (
+	schemeName = "googlesecretmanagerprovider"
+)
+
+type provider struct {
+	mu     sync.Mutex
```

Comment:
How come we need this? Can multiple of the same providers even exist in a single collector? I see the AWS provider doesn't have this

Reply:
One provider will exist in the Collector, but it appears that URIs are collected synchronously: 
https://github.com/open-telemetry/opentelemetry-collector/blob/5ee4816a4b3ef40f870a87e3cc4e2a15ab51b82d/confmap/resolver.go#L168

So I'm skeptical the mutex is required.

Reply:
Done